src/main/java/bio/terra/validation/BillingProfileRequestValidator.java

```diff
@@ -0,0 +1,36 @@
+package bio.terra.validation;
+
+import bio.terra.model.BillingProfileRequestModel;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import javax.validation.constraints.NotNull;
+import java.util.regex.Pattern;
+
+@Component
+public class BillingProfileRequestValidator implements Validator {
+
+    private static final String VALID_BILLING_ACCOUNT_ID_REGEX = "[A-Z0-9]{6}-[A-Z0-9]{6}-[A-Z0-9]{6}";
+
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return true;
+    }
+
+    @Override
+    public void validate(@NotNull Object target, Errors errors) {
+        if (target != null && target instanceof BillingProfileRequestModel) {
+            BillingProfileRequestModel billingProfileRequestModel = (BillingProfileRequestModel) target;
+            String billingAccountId = billingProfileRequestModel.getBillingAccountId();
+            if (!isValidAccountId(billingAccountId)) {
+                errors.rejectValue("billingAccountId",
+                    "The id must be 3 sets of 6 capitalized alphanumeric characters separated by dashes");
```

Comment:
We could up-case them.

Reply:
These are always presented up-case in the Google APIs/UIs -- I'd prefer to leave this expectation in.